src/Idris/Resugar.idr

```diff
@@ -47,7 +47,7 @@ mkOp tm@(PApp fc (PApp _ (PRef opFC kn) x) y)
        -- that the fixity context contains the name `(++)`
        let rootName = UN (Basic (nameRoot raw))
        let asOp = POp fc opFC kn (unbracketApp x) (unbracketApp y)
-       if not (null (lookupName rootName (infixes syn)))
+       if not (null (lookupName rootName (fixities syn)))
```

Comment:
What if we have a (private prefix) (export definition) in one module and a similar
(export infix) (export definition) in another? When resugaring aren't we going to
resugar the prefix op as an infix one here? That would be really quite confusing.

Reply:
Good point, let me see if I can create a test case for that

Reply:
I can confirm prefix `!!` in `!! a` is desugared as `a !!`. In the previous version, it was desugared as `(!!) a`. I'll work on a fix

Reply:
didn't have the time to look at this until now but using `infixes` instead of `fixities` solved it